### PR TITLE
[crunchyroll:beta] Handle empty episode/series titles

### PR DESCRIPTION
**Pull request by tejing1**
### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

I recently encountered a video on crunchyroll with an empty string for a title. Not all the metadata has been set for the video yet, apparently. yt-dlp fell back to the generic extractor for it, which was obviously not ideal. This PR also allows the same thing to happen with series titles.

### CODE REVIEW

File: yt_dlp/extractor/crunchyroll.py [hunk 1]
```diff
@@ -700,7 +700,7 @@ def _real_extract(self, url):
 
 class CrunchyrollBetaIE(CrunchyrollBaseIE):
     IE_NAME = 'crunchyroll:beta'
-    _VALID_URL = r'https?://beta\.crunchyroll\.com/(?P<lang>(?:\w{1,2}/)?)watch/(?P<internal_id>\w+)/(?P<id>[\w\-]+)/?(?:\?|$)'
+    _VALID_URL = r'https?://beta\.crunchyroll\.com/(?P<lang>(?:\w{1,2}/)?)watch/(?P<internal_id>\w+)/(?P<id>[\w\-]*)/?(?:\?|$)'
```
Comment (pukkandan) [inline yt_dlp/extractor/crunchyroll.py:703]: better to rename the groups as `id` and `display_id`

**Comment by pukkandan** [comment]
also, add tests

File: yt_dlp/extractor/crunchyroll.py (hunk 1 above)
Comment (tejing1) [reply yt_dlp/extractor/crunchyroll.py:703]: Yeah, that should get changed, but it seems like it fits better as part of my upcoming beta playlist support PR. Mind if I put it off till then?

**Comment by tejing1** [comment]
> also, add tests

- If you mean me to add a test that specifically shows the empty title case, I can't. The video in question has already had its metadata updated. But since it seems likely this will keep happening from time to time, I just thought it shouldn't prevent yt-dlp from downloading the video when in that state.
- If you mean add more tests generally, I can do that, but I'm not sure how much use they'll be given that the entire beta site is only accessible to logged in users currently. Basically you'd just be testing the code that redirects to the non-beta site when you're not logged in.

**Comment by pukkandan** [comment]
I meant this:
```py
{
    'url': 'https://beta.crunchyroll.com/watch/GY2P1Q98Y',
    'only_matching': True,
}
```

File: yt_dlp/extractor/crunchyroll.py (hunk 1 above)
Comment (pukkandan) [reply yt_dlp/extractor/crunchyroll.py:703]: Better to handle the whole thing in that PR then imho. Due to the huge scope of youtube-dl, if we make a PR of each small change, we'll never get anywhere. Feel free to make multiple changes in one PR as long as they concern the same site/extractor

Comment (tejing1) [reply yt_dlp/extractor/crunchyroll.py:703]: Alright. Will do.
